javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -1578,4 +1580,15 @@ protected void generateSendRequestSync(ClientMethod clientMethod, JavaType typeB
             function.methodReturn("this.sendRequestAsync(httpRequest).contextWrite(c -> c.putAll(FluxUtil.toReactorContext(context).readOnly())).block()");
         });
     }
+
+    private String getServiceVersionValue(ClientMethod clientMethod) {
+        String serviceVersion = "null";
+        if (JavaSettings.getInstance().isDataPlaneClient() && clientMethod.getProxyMethod() != null && clientMethod.getProxyMethod().getParameters() != null) {
+            if (clientMethod.getProxyMethod().getParameters().stream()
+                    .anyMatch(p -> p.getRequestParameterLocation() == RequestParameterLocation.QUERY && "apiVersion".equals(p.getName()))) {
```

Comment:
Can you use this? `ParameterSynthesizedOrigin.fromValue(parameter.getOrigin()) == ParameterSynthesizedOrigin.API_VERSION`